spdx-utils/src/main/antlr/com/here/ort/spdx/SpdxExpression.g4

```diff
@@ -83,7 +83,8 @@ OPEN  : '(' ;
 CLOSE : ')' ;
 PLUS  : '+' ;
 
-LICENSEREFERENCE : ('DocumentRef-' IDSTRING ':')? 'LicenseRef-' IDSTRING ;
-IDSTRING         : (ALPHA | DIGIT)(ALPHA | DIGIT | '-' | '.')* ;
+IDSTRING          : (ALPHA | DIGIT)(ALPHA | DIGIT | '-' | '.')* ;
+LICENSE_EXCEPTION : (IDSTRING '-exception' | IDSTRING '-exception-' IDSTRING) ;
+LICENSE_REFERENCE : ('DocumentRef-' IDSTRING ':')? 'LicenseRef-' IDSTRING ;
```

Comment:
The more specific `LICENSE_EXCEPTION` and `LICENSE_REFERENCE` will never be used, because the more general `IDSTRING` matches everything that these two also match, but is checked earlier if you put it first:
https://github.com/antlr/antlr4/blob/master/doc/faq/lexical.md#why-are-my-keywords-treated-as-identifiers

Reply:
Thanks for the explanation.